### ISSUE TITLE
progress bar - move aria attributes

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-next#1456]
The progress bar currently relies on `aria-describedby` to announce the title/description for the progress bar, but this attribute isn't reliably supported for things other than form elements (see [Short note on aria-label, aria-labelledby, and aria-describedby](https://developer.paciellogroup.com/blog/2017/07/short-note-on-aria-label-aria-labelledby-and-aria-describedby/)). The [aria documentation](https://support.freedomscientific.com/Content/Documents/Other/JAWS-ARIA-Support.doc) provided by JAWS also provides no indication that it intends to support this attribute for anything other than form elements: 
> JAWS announces this state for form controls. The user can read the describedby text using INSERT+ALT+R.

To make the progress bar title accessible, the aria attributes should move from the parent div to the div that displays the bar. 

This is what the html is today:
```
<div class="pf-c-progress" id="progress-simple-example" role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="33" aria-describedby="progress-simple-example-description">
  <div class="pf-c-progress__description" id="progress-simple-example-description">
    Title
  </div>
  <div class="pf-c-progress__status">
    <span class="pf-c-progress__measure">33%</span>
  </div>
  <div class="pf-c-progress__bar">
    <div class="pf-c-progress__indicator" style="width: 33%;">
    </div>
  </div>
</div>
```
This is what it should be:
```
<div class="pf-c-progress" id="progress-simple-example">
  <div class="pf-c-progress__description" id="progress-simple-example-description">
    Title
  </div>
  <div class="pf-c-progress__status" aria-hidden="true">
    <span class="pf-c-progress__measure">33%</span>
  </div>
  <div class="pf-c-progress__bar" role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="33"  aria-describedby="progress-simple-example-description">
    <div class="pf-c-progress__indicator" style="width: 33%;">
    </div>
  </div>
</div>
```
The difference is that:

- `aria-hidden="true"` is added to `.pf-c-progress__status` (this gets announced when `.pf-c-progress__bar` receives focus)
- `role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="33"` are moved to `.pf-c-progress__bar`
- Regarding `aria-describedby="progress-simple-example-description"`, it's debatable whether we should keep it or remove it. For screen readers that support it, it might be nice to restate the title of the progress bar. For now, I'm moving this to `.pf-c-progress__bar` as well.
- It's also debatable if more semantic elements should be used for `pf-c-progress` and `pf-c-progress__description` (like `<section>` and `<h*>` elements). There doesn't seem to be a good way to associate the title text with the progress bar except by proximity and using a heading element. Any suggestions here are welcome.
